bundles/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/internal/handler/NikoHomeControlAlarmHandler.java

```diff
@@ -0,0 +1,217 @@
+            // We need to do this in a separate thread because we may have to wait for the
+            // communication to become active
+            commStartThread = scheduler.submit(this::startCommunication);
+        }
+    }
+
+    @Override
+    synchronized void startCommunication() {
+        NikoHomeControlCommunication nhcComm = getCommunication(getBridgeHandler());
+
+        if (nhcComm == null) {
+            return;
+        }
+
+        if (!nhcComm.communicationActive()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "@text/offline.communication-error");
+            return;
+        }
+
+        NhcAlarm nhcAlarm = nhcComm.getAlarmDevices().get(deviceId);
+        if (nhcAlarm == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.configuration-error.deviceId");
+            return;
+        }
+
+        nhcAlarm.setEventHandler(this);
+
+        updateProperties(nhcAlarm);
+
+        String location = nhcAlarm.getLocation();
+        if (thing.getLocation() == null) {
+            thing.setLocation(location);
+        }
+
+        this.nhcAlarm = nhcAlarm;
+
+        initialized = true;
+        deviceInitialized();
+    }
+
+    @Override
+    void refresh() {
+        NhcAlarm alarm = nhcAlarm;
+        if (alarm != null) {
+            alarmEvent(alarm.getState());
+        }
+    }
+
+    @Override
+    public void dispose() {
+        NikoHomeControlCommunication nhcComm = getCommunication(getBridgeHandler());
+        if (nhcComm != null) {
+            NhcAlarm alarm = nhcComm.getAlarmDevices().get(deviceId);
+            if (alarm != null) {
+                alarm.unsetEventHandler();
+            }
+        }
+        nhcAlarm = null;
+        super.dispose();
+    }
+
+    private void updateProperties(NhcAlarm nhcAlarm) {
+        Map<String, String> properties = new HashMap<>();
+
+        if (nhcAlarm instanceof NhcAlarm2) {
+            NhcAlarm2 alarm = (NhcAlarm2) nhcAlarm;
+            properties.put(PROPERTY_DEVICE_TYPE, alarm.getDeviceType());
+            properties.put(PROPERTY_DEVICE_TECHNOLOGY, alarm.getDeviceTechnology());
+            properties.put(PROPERTY_DEVICE_MODEL, alarm.getDeviceModel());
+        }
+
+        thing.setProperties(properties);
+    }
+
+    @Override
+    public void alarmEvent(String state) {
+        NhcAlarm nhcAlarm = this.nhcAlarm;
+        if (nhcAlarm == null) {
+            logger.debug(" alarm device with ID {} not initialized", deviceId);
+            return;
+        }
+
+        updateState(CHANNEL_CONTROL,
+                NHCOFF.equals(state) || NHCDETECTORPROBLEM.equals(state) ? OnOffType.OFF : OnOffType.ON);
+        updateState(CHANNEL_ARMED,
+                NHCOFF.equals(state) || NHCPREARMED.equals(state) || NHCDETECTORPROBLEM.equals(state) ? OnOffType.OFF
+                        : OnOffType.ON);
```

Comment:
Only suggestion: use OnOffType.from

Reply:
I don't see how as some of these don't match on/off or true/false.

Reply:
`My idea: updateState(CHANNEL_CONTROL, OnOffType.from(NHCOFF.equals(state) || NHCDETECTORPROBLEM.equals(state)));`

Reply:
That’s a possibility. This is probably personal, but for me that does not look clearer. So I am not inclined to change it.